bin/zhtautau/runZHTauTauAnalysis.cc

```diff
@@ -1096,34 +1096,37 @@ int main(int argc, char* argv[])
       passLooseLepton &= lid==11 ? patUtils::passIso(leptons[ilep].el,  patUtils::llvvElecIso::Loose, patUtils::CutVersion::CutSet::ICHEP16Cut) :
       patUtils::passIso(leptons[ilep].mu,  patUtils::llvvMuonIso::Loose, patUtils::CutVersion::CutSet::ICHEP16Cut);
 
-
-
+      // passVeryLooseLepton
+      passVeryLooseLepton &= lid==11 ?  patUtils::passIso(leptons[ilep].el,  patUtils::llvvElecIso::VeryLoose, patUtils::CutVersion::CutSet::ICHEP16Cut) :
+      patUtils::passIso(leptons[ilep].mu,  patUtils::llvvMuonIso::VeryLoose, patUtils::CutVersion::CutSet::ICHEP16Cut);
+      
 
       //apply muon corrections
       //if(abs(lid)==13 && passIso && passId)
-      if(abs(lid)==13 && passLooseLepton){
+      if(abs(lid)==13 && passVeryLooseLepton){
         passSoftMuon=false;
-          if(is2016MC || is2016data){
-                   if(muCor2016){
-                     float qter =1.0;
-                     TLorentzVector p4(leptons[ilep].px(),leptons[ilep].py(),leptons[ilep].pz(),leptons[ilep].energy());
-                     int ntrk = leptons[ilep].mu.innerTrack()->hitPattern().trackerLayersWithMeasurement();
-                     if(is2016MC){muCor2016->momcor_mc  (p4, lid<0 ? -1 :1, ntrk, qter);
-                     }else if (is2016data){   muCor2016->momcor_data(p4, lid<0 ? -1 :1, 0, qter);
-                     }
-
-
-                     muDiff -= leptons[ilep].p4();
-                     leptons[ilep].setP4(LorentzVector(p4.Px(),p4.Py(),p4.Pz(),p4.E() ) );
-                     muDiff += leptons[ilep].p4();
-                   }
-          }
-        }
-
+	if(is2016MC || is2016data){
+	  if(muCor2016){
+	    float qter =1.0;
+	    TLorentzVector p4(leptons[ilep].px(),leptons[ilep].py(),leptons[ilep].pz(),leptons[ilep].energy());
+	    int ntrk = leptons[ilep].mu.innerTrack()->hitPattern().trackerLayersWithMeasurement();
+	    if(is2016MC){
+	      muCor2016->momcor_mc  (p4, lid<0 ? -1 :1, ntrk, qter);
+	    } else if (is2016data){   
+	      muCor2016->momcor_data(p4, lid<0 ? -1 :1, 0, qter);
+	    }
+	    
+	    
+	    muDiff -= leptons[ilep].p4();
+	    leptons[ilep].setP4(LorentzVector(p4.Px(),p4.Py(),p4.Pz(),p4.E() ) );
+	    muDiff += leptons[ilep].p4();
```

Comment:
I have already added that line but in my branch. 
https://github.com/clacaputo/2l2v_fwk/blob/80X_Fixing/bin/zhtautau/runZHTauTauAnalysis.cc#L1010

I forgot to push it in the central repository